examples/energy-management-app/energy-management-common/include/DEMManufacturerDelegate.h

```diff
@@ -0,0 +1,104 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+using chip::Protocols::InteractionModel::Status;
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace DeviceEnergyManagement {
+
+class DeviceEnergyManagementDelegate;
+
+/**
+ * Class to abstract manufacturer specific functionality
+ */
+class DEMManufacturerDelegate
+{
+public:
+    DEMManufacturerDelegate()
+    {
+    }
+
+    virtual ~DEMManufacturerDelegate()
+    {
+    }
+
+    virtual int64_t GetEnergyUse() = 0;
```

Comment:
Not clear what this one does - I can guess the others - but not sure if adding docstrings to this file would be helpful?
No point if the are obvious - comments can get stale and out of date with the code etc.

Reply:
The `PowerAdjustEnd `event has a field called `energyUse`. This is set in `DeviceEnergyManagementDelegate::SendPowerAdjustEndEvent`. Hence the need for the `DEMManufacturerDelegate `object which (should) know this value to return it.